### ISSUE TITLE
Create leaderboard for top cases

### ISSUE THREAD

**Issue by mlissner** [freelawproject/courtlistener#3036]
This has been a recurring concept over the years, since we're often aware of the popular legal documents of the moment. 

The first step would be to create some simple html dashboards with the top downloads. 

Next, we add (paid?) newsletters and Twitter bots, like Hacker News has, that share the top N items for the last day or week or whatever. 

A few thoughts:

1. Redis has a great tool for this called TOPK: https://redis.com/blog/meet-top-k-awesome-probabilistic-addition-redis/

That can probably be the underlying algorithm. 

2. Doing this for many topics or making it possible to do custom search-based leaderboards would be killer. Imagine making a search query and getting the top items for the query each day or week.  

3. Architecturally, it's possible to include this in CourtListener, or it could work to put it under the free.law domain. (news.free.law? top.free.law?)

4. Getting stats from S3 will be annoying, but perhaps Athena is the trick to query our logs periodically, or perhaps we use a CloudFront lambda.

**Comment by cweider** [comment]
A more ready alternative could be plausible.io’s [Stats API](https://plausible.io/docs/stats-api)? I’ve no experience _this_ analytics framework, but the documentation makes it sound plausible. A line of work that I could expect coming out of this: definition of event names, properties, and triggers to make certain things queryable.

(Aside: analytics can also be very useful for informing website design – e.g. finding common usage patterns)

**Comment by mlissner** [comment]
> A more ready alternative could be plausible.io’s [Stats API](https://plausible.io/docs/stats-api)?

Yeah, maybe. I actually think the TOPK approach looks pretty easy. It has three commands. One to make the data structure, another to add to it, and a third to get its results. There are some knobs to tweak when creating the item, but other than that, it looks pretty simple, I think. One issue with Plausible is that it charges by the hit. We already only log 1 of every ten requests to save on our Plausible bill, and that doesn't even include the PDF downloads, which can be a million hits in 24 hours....

**Comment by cweider** [comment]
Oooh, much less of a good deal than sentry – yea, that data warehousing cost would become outrageous.
